### ISSUE TITLE
step 90 Build a Music Player (JS Beta) last instruction refers to attributes not properties of an element

### ISSUE THREAD

**Issue by hbar1st** [freeCodeCamp/freeCodeCamp#55392]
### Describe the Issue

In the last instruction given of step 90 of the build a music player project (JS Beta), we are asked to:
>Set the id attribute of resetButton to "reset" and its "aria-label" attribute to "Reset playlist".

My quibble with this is that resetButton doesn't have a 'aria-label' attribute, it has a 'ariaLabel' property (same for id). Shouldn't this last step say 'property' instead of 'attribute' (twice) and 'ariaLabel' instead of 'aria-label'?
(I understand that the equivalent html element has these attributes but I just feel that we shouldn't be so cavalier with our terms)

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-90

### Your code

N/A

### Expected behavior

N/A

### System

N/A

**Comment by hbar1st** [comment]
It is true that “confusion” is going to happen. 😉 Also true that the hint or the first line will address the confusion. My thought is: why cause that confusion at all?

**Comment by lasjorg** [comment]
I don't really mind it. They are first taught the connection between attributes and element properties. Then asked to set the attributes. I wouldn't call it confusing, just forcing you to "connect the dots".

As for the verbiage, it is correct to say you are setting attributes even when using the DOM interface properties. You are setting attributes by way of setting properties.

I mean, we could be more explicit about it if we want to. And seeing as we do not allow the use of `setAttribute`, "setting the attributes" isn't really the requirement, but "setting the attributes" using the element properties is.

**Comment by hbar1st** [comment]
I think I have a solid background in UXD which is why I tend to err on the side of the user. I would want to, for eg, check how often people fail on this step on the first try. I think if the majority of them do, we have to ask: are we trying to cause a stumble? I don't mind if we are, I just mind if we are doing it without any clear intent. I mean, there is something to be said for failure, it helps us remember, but it does bother me that the failure is not -meant- here. It just happens because of the terminology being used. (I'm assuming that many people are failing - would be interested to hear if I'm wrong)

**Comment by lasjorg** [comment]
> are we trying to cause a stumble?

I don't think we are. Just because they have to use element properties doesn't mean the paragraph...

> Set the `id` attribute of `resetButton` to `"reset"` and its `"aria-label"` attribute to `"Reset playlist"`.

...is wrong.

That is what they have to do, set the attributes using the element properties.

---

As I said, if we want to focus on the properties we can, I'm just not convince it is needed here.

But have you seen a lot of campers fail this step because of the requirements being worded the way they are? Because, if so, we might want to change it.

**Comment by hbar1st** [comment]
I don't have a way to check if people fail on the first try. The hint effectively gives the solution after the first failure so I highly doubt people struggle after they see it. If fCC keeps track of numbers of failures on first try, then that's what I would like to see personally.

I came across something else on the forum a few times which made me think of this issue. People stumble when they first encounter the instruction "print <something>' as they don't know what 'print' means till someone tells them it means console.log something. (they think it means to simply type something). The reason this made me think of this issue is because it is a matter of a word, a simple one for a developer, but not so simple for a newbie. The hint can be changed to give the answer (essentially like this step does) and that would 'fix' it, but I'd be just as dissatisfied with this as a solution because the step should have been clearer from the start. (and just for fun, I will now contradict myself by saying: I've often thought we spoon-feed the campers too much in the instructions. Some vagueness may be frustrating but ultimately makes for a better student? I just want the -intention- of being vague to be there from the start. We want to be vague, so they think for themselves? Cool. Or, we want to make them 'connect the dots', cool. Conversely, we want to be specific and leave nothing to the imagination, that's cool too. Whatever it is, fCC should just stick with one strategy all the way. Preferably this strategy is the mission statement somewhere for all curriculum steps, and every step is measured against that yardstick. Clear, simple and precise wording and instructions, yup. Or, more high-level wording that encourages inquiry, why not! let's pick one though?) That's the end of my soap-box preaching now. I appreciate everyone's time.

**Comment by lasjorg** [comment]
We shouldn't use "print" for JS. I do think "log" would be fine, at least for later challenges. I would suggest we change all instances of "print" to "log", "console log", or `console.log`.

But I don't think that is related to this issue.

The camper is taught how to set attributes through properties, then asked to set specific attributes. Sure, we could ask them to set the properties, but I think it is well implied by the previous instructions. I'm not sure how else they would set the attributes?

I do think the requirements as they are worded now reinforces something valuable, namely that attributes can be set through element properties. And that connection between the attributes and properties might get lost if we change the language of the requirements.

We should use a consistent, technical, correct language. But further into the curriculum, the less handholding is needed, and some implied knowledge is expected. This project is pretty far into the curriculum, so first teaching about setting attributes through properties, and then asking for attributes to be set, without explicitly mentioning the properties, seems fairly reasonable to me.

---

Just to be clear, I'm not wildly against changing the language. Even if I think it is fine as is.

**Comment by hbar1st** [comment]
I've been told I argue too much at times. I'll let other people chime in if they have an opinion one way or the other. I feel I'll just be repeating myself if I attempt to speak further.
I would be okay if we changed the text to:

Set the id attribute of resetButton element to "reset" and its "aria-label" attribute to "Reset playlist".

(i added the word element while changing nothing else, as to me, it is the element that has the attributes)

**Comment by lasjorg** [comment]
I would be fine with that change. I will open it up to contribution.

---

This looks like something that can be fixed by "first time" code contributors to this repository. Here are the files that you should be looking at to work on a fix:

File to change:

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/655729e68e49b277a6b448bd.md

Change the text:
```md
Set the `id` attribute of `resetButton` to `"reset"` and its `"aria-label"` attribute to `"Reset playlist"`.
```

To:

```md
Set the `id` attribute of the `resetButton` element to `"reset"` and its `"aria-label"` attribute to `"Reset playlist"`.
```

Please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/), we prioritize contributors following the instructions in our guides. Join us in our chat room or the forum if you need help contributing, our moderators will guide you through this.

Sometimes we may get more than one pull-requests. We typically accept the most quality contribution followed by the one that is made first.

Happy contributing.

**Comment by shah-azman-sakin-alam** [comment]
Well in javascript its like ariaLabel instead of aria-label  and in html its a attribute and in javascript when you set the attribute using DOM, imagine like for example in "element.ariaLabel" element is a object, and ariaLabel is a property of the object, you set the attribute in, so like in html there's no objects or anything like that but in html files when you set aria-label, and you access a element from dom, the aria-label is presented something like this not exactly but somewhat

element{
   ariaLabel: //aria label here
}
